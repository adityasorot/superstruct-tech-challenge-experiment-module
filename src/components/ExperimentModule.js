import { Box, Collapse, IconButton, Paper, Typography } from "@mui/material";
import React, { useState } from "react";
import LockIcon from "@mui/icons-material/Lock";
import LockOpenIcon from "@mui/icons-material/LockOpen";
import ExperimentModuleCollapse from "./ExperimentModuleCollapse";
const ExperimentModule = () => {
  const [iterationData, setIterationData] = useState([]);
  const [moduleLockState, setModuleLockState] = useState(false);
  const [moduleOpenState, setModuleOpenState] = useState(true);

  const lockButtonHandler = (e) => {
    setModuleLockState(!moduleLockState);
    setModuleOpenState(false);
    e.stopPropagation();
  };
  const mainButtonHandler = () => {
    if (!moduleLockState) setModuleOpenState(!moduleOpenState);
  };

  return (
    <Paper
      sx={{
        backgroundColor: "#222222",
        color: "white",
        py: 4,
        px: 2,
        m: 3,
        borderRadius: 3,
      }}
      elevation={3}
    >
      <Box
        sx={{ display: "flex", flexDirection: "row", paddingBottom: 3 }}
        onClick={mainButtonHandler}
      >
        <Typography
          variant="h4"
          sx={{ flexGrow: 1, color: moduleLockState ? "#757575" : "white" }}
        >
          Experiment Module
        </Typography>
        {iterationData.length > 0 ? (
          <IconButton onClick={lockButtonHandler}>
            {moduleLockState ? (
              <LockIcon sx={{ color: "#757575" }} />
            ) : (
              <LockOpenIcon sx={{ color: "#757575" }} />
            )}
          </IconButton>
        ) : (
          <></>
        )}
      </Box>
      <Collapse in={moduleOpenState} unmountOnExit>
        <ExperimentModuleCollapse
          iterationData={iterationData}
          setIterationData={setIterationData}
          lockButtonHandler={lockButtonHandler}
        />
      </Collapse>
    </Paper>
  );
};

export default ExperimentModule;
