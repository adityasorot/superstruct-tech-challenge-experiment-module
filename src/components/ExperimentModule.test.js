import renderer from "react-test-renderer";
import ExperimentModule from "./ExperimentModule";

it("Experiment Module", () => {
  const component = renderer.create(<ExperimentModule />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
