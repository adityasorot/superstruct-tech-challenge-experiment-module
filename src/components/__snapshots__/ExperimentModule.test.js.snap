// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiment Module 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-1efqb74-MuiPaper-root"
>
  <div
    className="MuiBox-root css-7ohv8l"
    onClick={[Function]}
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 css-h7wn88-MuiTypography-root"
    >
      Experiment Module
    </h4>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
          style={
            Object {
              "minHeight": "0px",
            }
          }
        >
          <div
            className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                className="MuiBox-root css-c6l12j"
              >
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-11gc9ze-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  <span
                    className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
                  >
                    <svg
                      aria-hidden={true}
                      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="AddIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  Add Iteration
                </button>
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  Reset
                </button>
                <button
                  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                  disabled={false}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex={0}
                  type="button"
                >
                  Lock
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
